plugins/reports/monographReport/MonographReportPlugin.php

```diff
@@ -0,0 +1,156 @@
+
+use APP\author\Author;
+use APP\core\Request;
+use APP\facades\Repo;
+use APP\press\Press;
+use APP\press\Series;
+use APP\press\SeriesDAO;
+use APP\publication\Publication;
+use APP\publicationFormat\IdentificationCode;
+use APP\publicationFormat\PublicationFormat;
+use APP\submission\Collector;
+use APP\submission\Submission;
+use Exception;
+use PKP\category\Category;
+use PKP\db\DAORegistry;
+use PKP\plugins\ReportPlugin;
+use SplFileObject;
+
+class MonographReportPlugin extends ReportPlugin
+{
+    /**
+     * @copydoc Plugin::register()
+     *
+     * @param null|mixed $mainContextId
+     */
+    public function register($category, $path, $mainContextId = null): bool
+    {
+        $success = parent::register($category, $path, $mainContextId);
+        $this->addLocaleData();
+        return $success;
+    }
+
+    /**
+     * @copydoc Plugin::getName()
+     */
+    public function getName(): string
+    {
+        return substr(static::class, strlen(__NAMESPACE__) + 1);
+    }
+
+    /**
+     * @copydoc Plugin::getDisplayName()
+     */
+    public function getDisplayName(): string
+    {
+        return __('plugins.reports.monographReport.displayName');
+    }
+
+    /**
+     * @copydoc Plugin::getDescription()
+     */
+    public function getDescription(): string
+    {
+        return __('plugins.reports.monographReport.description');
+    }
+
+    /**
+     * @copydoc ReportPlugin::display()
+     */
+    public function display(array $args, Request $request): void
+    {
+        $press = $request->getContext();
+        if (!$press) {
+            throw new Exception('The monograph report requires a context');
+        }
+
+        header('content-type: text/comma-separated-values');
+        header('content-disposition: attachment; filename=monograph-report-' . date('Ymd') . '.csv');
+
+        $fields = $this->_getFieldMapper($press, $request);
+        $output = new SplFileObject('php://output', 'w');
+        try {
+            // UTF-8 BOM to force the file to be read with the right encoding
+            $output->fwrite("\xEF\xBB\xBF");
+            $output->fputcsv(array_keys($fields));
+
+            $monographs = Repo::submission()->getCollector()
+                ->filterByContextIds([$press->getId()])
+                ->filterByStatus([Submission::STATUS_PUBLISHED])
+                ->orderBy(Collector::ORDERBY_DATE_SUBMITTED, Collector::ORDER_DIR_ASC)
+                ->getMany();
+
+            /** @var Submission */
+            foreach ($monographs as $monograph) {
+                $output->fputcsv(array_map(fn (callable $getter) => $getter($monograph->getCurrentPublication()), $fields));
+            }
+        } finally {
+            $output = null;
+        }
+    }
+
+    /**
+     * Retrieves a dictionary, where the keys represent the column titles and the values a value getter
+     */
+    private function _getFieldMapper(Press $press, Request $request): array
```

Comment:
We do not ('need' to) use underscores any more, I think...

Reply:
I'm not sure as well haha, I like the prefix, it makes it easier for me to understand whether something is private/public, but I'll double check our rules/latest code :D